js/content.js

```diff
@@ -0,0 +1,447 @@
+  item.dispatchEvent(new MouseEvent('mousedown', {bubbles: true}));
+  item.dispatchEvent(new PointerEvent('pointerup', {bubbles: true}));
+  item.dispatchEvent(new MouseEvent('mouseup', {bubbles: true}));
+  item.dispatchEvent(new MouseEvent('mouseout', {bubbles: true}));
+  item.dispatchEvent(new MouseEvent('click', {bubbles: true}));
+  item.dispatchEvent(new Event('change', {bubbles: true}));
+  return true;
+}
+
+function changeView(viewIndex) {
+	if (5 <= viewIndex <= 8) {
+		var gameframe = document.getElementsByClassName('gameframe')[0];
+		gameframe.contentWindow.document.querySelector('[data-hook="settings"]').click();
+		var viewModeToggle = waitForElement('[data-hook="viewmode"]')
+		viewModeToggle.then(function (toggle) {
+			toggle.selectedIndex = viewIndex;
+			simulateClick(toggle);
+			closeBtn = waitForElement('[data-hook="close"]');
+			closeBtn.then(function (btn) { btn.click() })
+		})
+	}
+}
+
+function record(gameview = true) {
+	var gameframe = document.getElementsByClassName('gameframe')[0];
+	if (gameview) {
+		gameframe.contentWindow.document.querySelector('[data-hook="menu"]').click();
+		var recBtn = waitForElement('[data-hook="rec-btn"]');
+		recBtn.then(function (btn) { btn.click() });
+		gameframe.contentWindow.document.querySelector('[data-hook="menu"]').click();
+	}
+	else {
+		gameframe.contentWindow.document.querySelector('[data-hook="rec-btn"]').click();
+	}
+}
+
+chatObserver = new MutationObserver(function(mutations) {
+	var candidates = mutations.flatMap(x => Array.from(x.addedNodes)).filter(x => x.tagName == 'P');
+	var gameframe = document.documentElement.getElementsByClassName("gameframe")[0];
+	var bottomSec = gameframe.contentWindow.document.getElementsByClassName('bottom-section')[0];
+	var statSec = gameframe.contentWindow.document.getElementsByClassName('stats-view')[0];
+	var chatInput = gameframe.contentWindow.document.querySelector('[data-hook="input"]');
+	var chatLog = gameframe.contentWindow.document.querySelector('[data-hook="log"]');
+	
+	chatCheck = function(chatLine) {
+		if ([...muted].filter(x => chatLine.innerText.startsWith(x + ': ')).length > 0) {
+			chatLine.hidden = true;
+		}
+		else if (muteAllToggle && muteExceptions.filter(x => chatLine.innerText.startsWith(x + ': ')) == 0 && chatLine.className != 'notice') {
+			chatLine.hidden = true;
+		}
+		
+		if (chatLine.innerText.startsWith('Game start')) {
+			toggleChatOpt();
+			toggleChatKb();
+		}
+		
+		chrome.storage.local.get({'haxTransChatConfig' : false},
+			function (items) {
+				if (items.haxTransChatConfig) { 
+					if (chatLine.innerText.startsWith('Game start')) {	
+						chatFormat(bottomSec,statSec,chatInput,'absolute');
+					}
+					else if (chatLine.innerText.startsWith('Game stop')) {	
+						bottomSec.removeAttribute('style');
+					}
+				}
+		});
+		
+		chatLine.innerHTML = linkify(chatLine.innerHTML);
+
+		// right click to tag
+		chatLine.oncontextmenu = function () {
```

Comment:
Need to incorporate latest change from my master branch, replaces nickname spaces with underscores + block for notice